jekyll/_cci2/language-android.md

```diff
@@ -133,12 +147,107 @@ and macOS capabilities. Please check out [this example React Native
+and clicking **Enable API**.
+
+In your `.circleci/config.yml` file,
+add the following `run` steps.
+
+1. **Build the debug APK and test APK.**
+Use Gradle to build two APKs.
+To improve build performance,
+consider [disabling pre-dexing](#disabling-pre-dexing).
+
+2. **Store the service account.**
+Store the service account you created in a local JSON file.
+
+3. **Authorize `gcloud`**.
+Authorize the `gcloud` tool
+and set the default project.
+
+4. **Use `gcloud` to test with Firebase Test Lab.**
+Adjust the paths to the APK files
+to correspond to your project.
+
+5. **Install `crcmod` and use `gsutil` to copy test results data.**
+`crcmod` is required
+to use `gsutil`.
+Use `gsutil`
+to download the newest files in the bucket to the CircleCI artifacts folder.
+Be sure to replace `BUCKET_NAME` and `OBJECT_NAME` with project-specific names.
+
+```yaml
+version: 2
+jobs:
+  test:
+    docker:
+      - image: circleci/android:api-28-alpha  # gcloud is baked into this image
+    steps:
+      - run:
+          name: Build debug APK and release APK
+          command: |
+            ./gradlew :app:assembleDebug
+            ./gradlew :app:assembleDebugAndroidTest
+      - run:
+          name: Store Google Service Account
+          command: echo $GCLOUD_SERVICE_KEY > ${HOME}/gcloud-service-key.json
+      - run:
+          name: Authorize gcloud and set config defaults
+          command: |
+            sudo gcloud auth activate-service-account --key-file=${HOME}/gcloud-service-key.json
+            sudo gcloud --quiet config set project ${GOOGLE_PROJECT_ID}
+      - run:
+          name: Test with Firebase Test Lab
+          command: >
+            sudo gcloud firebase test android run
+              --app <local_server_path>/<app_apk>.apk
+              --test <local_server_path>/<app_test_apk>.apk
+              --results-bucket cloud-test-${GOOGLE_PROJECT_ID}
+      - run:
+          name: Install gsutil dependency and copy test results data
+          command: |
+            sudo pip install -U crcmod
+            sudo gsutil -m cp -r -U `sudo gsutil ls gs://[BUCKET_NAME]/[OBJECT_NAME] | tail -1` ${CIRCLE_ARTIFACTS}/ | true
+```
+
+For more details on using `gcloud` to run Firebase,
+see the [official documentation](https://firebase.google.com/docs/test-lab/android/command-line).
 
-Disabling pre-dexing for Android builds on CircleCI can speed up your builds. Refer to the [Disable Android pre dexing on CI builds](http://www.littlerobots.nl/blog/disable-android-pre-dexing-on-ci-builds/) blog post for details.
+## Disabling Pre-Dexing
 
-CircleCI always runs clean builds, so pre-dexing has no benefit. By default, the Gradle Android plugin pre-dexes dependencies. Pre-dexing converts Java bytecode into Android bytecode to speed up development by doing only incremental dexing as you change code. Pre-dexing can make compilation slower and may also use large quantities of memory. 
```

Comment:
Here, information ordering has been altered for better flow.

1. Introduce Android's default behavior and why it's usually good.
2. Say why it's not good on CircleCI.
3. Say what you should do instead.

Reply:
I disagree, we should tell the user what to do and why first, then give the background info about Android.